platforms/documentation/docs/src/docs/userguide/authoring-builds/tasks/more_about_tasks.adoc

```diff
@@ -81,180 +69,161 @@ include::sample[dir="snippets/tasks/defineAsKotlinDelegatedProperty/groovy",file
-include::sample[dir="snippets/tasks/defineAndConfigure/groovy",files="build.gradle[tags=no-description]"]
+include::sample[dir="snippets/files/archiveNaming/kotlin",files="build.gradle.kts[tags=zip-task-with-custom-base-name]"]
+include::sample[dir="snippets/files/archiveNaming/groovy",files="build.gradle[tags=zip-task-with-custom-base-name]"]
 ====
 
-.Don't forget about the build phases
---
-A task has both configuration and actions. When using the `doLast`, you are simply using a shortcut to define an action. Code defined in the configuration section of your task will get executed during the configuration phase of the build regardless of what task was targeted. See <<build_lifecycle.adoc#build_lifecycle,Build Lifecycle>> for more details about the build lifecycle.
---
+The `myCustomZip` task will ZIP the files in the _configuration phase_ and the name of the archive will be printed in the _execution phase_.
 
 [[sec:passing_arguments_to_a_task_constructor]]
-== Passing arguments to a task constructor
+== Passing arguments
 
 As opposed to configuring the mutable properties of a `Task` after creation, you can pass argument values to the `Task` class's constructor.
-In order to pass values to the `Task` constructor, you must annotate the relevant constructor with `@javax.inject.Inject`.
 
-.Task class with `@Inject` constructor
+In order to pass values to the `Task` constructor, you must annotate the relevant constructor with `@javax.inject.Inject`:
+
 ====
 include::sample[dir="snippets/tasks/taskConstructorArgs-onTaskContainer/kotlin",files="build.gradle.kts[tags=inject-task-constructor]"]
 include::sample[dir="snippets/tasks/taskConstructorArgs-onTaskContainer/groovy",files="build.gradle[tags=inject-task-constructor]"]
 ====
 
-You can then create a task, passing the constructor arguments at the end of the parameter list.
+Then, you can create a task by passing the constructor arguments at the end of the parameter list:
 
-.Registering a task with constructor arguments using TaskContainer
 ====
 include::sample[dir="snippets/tasks/taskConstructorArgs-onTaskContainer/kotlin",files="build.gradle.kts[tags=on-task-container]"]
 include::sample[dir="snippets/tasks/taskConstructorArgs-onTaskContainer/groovy",files="build.gradle[tags=on-task-container]"]
 ====
 
-NOTE: It's recommended to use the <<task_configuration_avoidance.adoc#task_configuration_avoidance,Task Configuration Avoidance>> APIs to improve configuration time.
-
-In all circumstances, the values passed as constructor arguments must be non-null.
+The values passed as constructor arguments must be non-null.
 If you attempt to pass a `null` value, Gradle will throw a `NullPointerException` indicating which runtime value is `null`.
 
+TIP: The <<task_configuration_avoidance.adoc#task_configuration_avoidance,Task Configuration Avoidance>> APIs are recommended as they improve configuration time.
```

Comment:
🤔 That tip feels a bit out of place now that the whole page is written with configuration avoidance in mind.